### ISSUE TITLE
Another Fatal Design Flaw

### ISSUE THREAD

**Issue by rusty555** [PACTCare/Dweb.page#3]
I notice that your security model is to encrypt a file, store it _publicly_ on the IPFS network and then share the secret key. Please refer to the following discussion with the founder of IPFS and the recommendation to design a capability system instead:
https://github.com/ipfs/ipfs/issues/86

Quote:
"Naive example: Encrypt some file with some secret, and share the secret with the people that have read permission"

Secondly, recommending people to share private keys via WhatsApp or Telegram as suggested in your security white paper is also very naive. Private keys are not intended to ever be transmitted over a channel - even encrypted.

**Comment by Noc2** [comment]
Okay. First of all thanks for taking the time and creating a complete new GitHub account just to comment here and also for the productive and motivating title 😉. (you notice that I’m doing all this without being paid?) I still disagree that the first issue was even a design flaw when it mentions things that aren’t even integrated into Dweb.page. 
But you’re actually right here to some degree. Encryption is never 100% secure and especially if other people could create copies of the file and decrypt it in the future. I still believe it would be secure enough for me to send a file from my pc to a friend via Dweb.page. I would prefer it to almost all other centralized solutions or trusting the post office etc.!
If you want, we can test it, and you try to tell me more about my encrypted content. It would basically mean you intercepted my email because otherwise you probably have no idea what hash is actually the interesting hash and it might be gone before you can download it (unless you keep a copy of all hashes on the ipfs network, which might be awesome for a lot of people :-)). And then you end up with almost the same situation like end-to-end encrypted emails, but sure you can break AES256-GCM (if not now then definitely if you store this email for 50 years).
So to summarize, I want to improve the security of the tool and if you have ideas on how to do it (some are mentioned in the link you shared and seem to be investigated by ipfs), I will try to integrate them or hope that others help me with it! 
But please try to be more productive with your critique in the future, there might be other people out there which are working in their free time on open source projects and might quite this because of issues like the two "Fatal Design Flaw" here.

**Comment by Noc2** [comment]
I close this issue due to inactivity and because it’s quite general. Nevertheless, I will, of course, keep working on the security (see for example the new storing of non-extractable signature keys in indexeddb) as well as the support of the ipfs companion, which might be used to restrict the sharing of private files.
